### PR TITLE
[DD4hep] Add DD4hep Run 3 workflow that reads geometry from DB

### PR DESCRIPTION
**Pull request by cvuosalo**
Up to now, DD4hep workflows have read geometry from XML files. There is now a DD4hep ideal geometry description in the Conditions DB, so a DD4hep workflow that reads geometry from the DB can be created.

Addition to the PR/IB tests for this workflow is commented out for now until the Cond DB issue with creating payloads is resolved.

This PR depends on PR #34344 to run.

#### PR validation:
`runTheMatrix.py` correctly creates the configuration for running this workflow. The workflow can run after the Cond DB issue with creating DB payloads is resolved.

No backport will be done.

### CODE REVIEW

**Comment by cvuosalo** [comment]
When I try to run the new 11634.912 workflow, the following error occurs. I don't know what it means.
```
n exception of category 'ConditionDatabase' occurred while
   [0] Processing global begin Run 
Exception Message:
Payload of type FileBlob with id 2fdfa84a08623caa48cec3d71726922f938d2cd5 could not be loaded. An exception of category 'ConditionDatabase' occurred.
Exception Message:
De-serialization failed: invalid signature Payload serialization info:  {
"CMSSW_version": "CMSSW_12_0_X_2021-07-03-1100",
"architecture": "slc7_amd64_gcc900",
"technology": "boost/serialization",
"tech_version": "1_75"
 }
 from default_deserialize 
 from Session::fetchPayload 
```

**Comment by cvuosalo** [comment]
@civanch FYI: Here's the DD4hep DB workflow PR.

**Comment by srimanob** [comment]
> When I try to run the new 11634.912 workflow, the following error occurs. I don't know what it means.
> 
> ```
> n exception of category 'ConditionDatabase' occurred while
>    [0] Processing global begin Run 
> Exception Message:
> Payload of type FileBlob with id 2fdfa84a08623caa48cec3d71726922f938d2cd5 could not be loaded. An exception of category 'ConditionDatabase' occurred.
> Exception Message:
> De-serialization failed: invalid signature Payload serialization info:  {
> "CMSSW_version": "CMSSW_12_0_X_2021-07-03-1100",
> "architecture": "slc7_amd64_gcc900",
> "technology": "boost/serialization",
> "tech_version": "1_75"
>  }
>  from default_deserialize 
>  from Session::fetchPayload 
> ```

@cms-sw/alca-l2 Could you please advise? Thanks.

**Comment by cvuosalo** [comment]
It would be good to get this PR into 12_0_0_pre4.

**Comment by srimanob** [comment]
test parameters:

- workflow = 11634.912

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
Hi @civanch 
The workflow defined here is to make short matrix runs. All run-3 workflows with DB DD4hep can be done with upgrades option of runTheMatrix, or with relvals machine of PdmV. No need to define them here.

**Comment by srimanob** [comment]
Urgent

**Comment by srimanob** [comment]
@cvuosalo @civanch @cms-sw/alca-l2 
Could you please clarify if the issue is on GT or something else? I assume when @cvuosalo develops this PR, the test is done with the candidate GT, or this is the last thing to solve before we can run until the end?

Since this is the workflow level, if we are not ready (e.g. can't pass the PR test), we do not need to be merge. This is not a showstopper to validate DD4hep DB because PdmV can pick the cmsDriver to submit the relvals anyways.

What do you think? @cms-sw/pdmv-l2 @silviodonato @qliphy @perrotta

**Comment by cvuosalo** [comment]
@srimanob The current situation is that the workflow in this PR requires the new DD4hep ideal geometry description DB payload. I created and uploaded this payload to the Conditions DB, but there is a glitch in the AlCaDB infrastructure that gives the wrong signature to payloads as they are created (see error here https://github.com/cms-sw/cmssw/pull/34343#issuecomment-875086587). 
There are two possible solutions:
1. Fix the AlCaDB infrastructure so payloads can be created correctly and uploaded (see PR #34350).
2. Create the payload with 12_0_0_pre3, which works correctly.

I am working on the second option, but merging my code back to pre3 requires compilation of a huge part of CMSSW, and that is taking very long. I would hope to be able to correctly create the payload on Wednesday and get this workflow running. I will report back here with any progress.

**Comment by cvuosalo** [comment]
I have uploaded a payload to the DB that should be free of the signature problem we saw above. When the AlCaDB conveners attach it to the DD4hep Global Tag, then the workflow in this PR should be able to run.

**Comment by francescobrivio** [comment]
> I have uploaded a payload to the DB that should be free of the signature problem we saw above. When the AlCaDB conveners attach it to the DD4hep Global Tag, then the workflow in this PR should be able to run.

I have created a new candidate including the new pyaload provided by @cvuosalo :
`specialDD4hepTEST_120X_mcRun3_2021_realistic_v1_Candidate_2021_07_07_12_35_50`
(https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/specialDD4hepTEST_120X_mcRun3_2021_realistic_v1_Candidate_2021_07_07_12_35_50)

**Comment by cvuosalo** [comment]
The error still occurs. I think the problem is that when I uploaded, the new payload's contents were actually identical to that of the payload with the bad signature. Thus, nothing was changed; only the new tag was added pointing to the original payload with the bad signature.
Maybe if the payload with the bad signature were entirely deleted, then I could re-upload the payload with pre3 to create a new payload with a good signature.

**Comment by cvuosalo** [comment]
This PR could be merged in its current form to define the `912` workflows. When the Cond DB problem with creating payloads is resolved, then a new PR could uncomment the 11634.912 workflow in `relval_2017.py` to enable it to be run by PR and IB tests.

**Comment by qliphy** [comment]
> This PR could be merged in its current form to define the `912` workflows. 

@cms-sw/pdmv-l2 @cms-sw/upgrade-l2   Would you please have a review on this PR?  We are going to build 12_0_0_pre4 soon today, and if possible we can include this PR.

**Comment by civanch** [comment]
Hi all,

Carl validated this PR locally. Currently, there is no possibility to test it in a standard way. Also this PR does not bring any harm, because new WFs are not yet used.

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py [hunk 1]
```diff
@@ -87,7 +87,8 @@ def runSelected(opt):
                      10024.0, #2017 ttbar
                      10224.0, #2017 ttbar PU
                      10824.0, #2018 ttbar
-                     11634.911, #2021 DD4hep ttbar
+                     11634.911, #2021 DD4hep ttbar reading geometry from XML
+                     11634.912, #2021 DD4hep ttbar reading geometry from the DB
```
Comment (srimanob) [inline Configuration/PyReleaseValidation/scripts/runTheMatrix.py:91]: This should also be disable because short matrix will run .912 also if it is here.

**Comment by srimanob** [comment]
> Hi all,
> 
> Carl validated this PR locally. Currently, there is no possibility to test it in the standard way. Also this PR does not bring any harm, because new WFs are not yet used.

If we would like to merge, we need to disable .912 also in `Configuration/PyReleaseValidation/scripts/runTheMatrix.py`

**Comment by chayanit** [comment]
Sure I will sign after .912 is disable in short matrix

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py (hunk 1 above)
Comment (cvuosalo) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: @srimanob @chayanit 912 is commented out now in the short matrix.

**Comment by srimanob** [comment]
Please test

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py (hunk 1 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: Thanks @cvuosalo

**Comment by srimanob** [comment]
> -1
> 
> **Failed Tests**: RelVals
> **Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-1112a1/16601/summary.html
> **COMMIT**: [05beb78](https://github.com/cms-sw/cmssw/commit/05beb78f45a507ef67abd6bb37ab16ec3f2f3e29)
> **CMSSW**: CMSSW_12_0_X_2021-07-07-2300/slc7_amd64_gcc900
> **User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34343/16601/install.sh` to create a dev area with all the needed externals and cmssw changes.
> 
> ## RelVals

This fails because I set .912 workflow before as test parameter of the workflow. Trying to delete it.

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
Please abort

**Comment by srimanob** [comment]
test parameter

- workflow =

**Comment by srimanob** [comment]
test parameters:

- workflow =

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
Please abort

**Comment by perrotta** [comment]
test parameters:

**Comment by srimanob** [comment]
test parameters:

- workflow = 11634.911

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
> test parameters:

Ah, OK. Thanks @perrotta

**Comment by srimanob** [comment]
Please abort

**Comment by srimanob** [comment]
test parameters:

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
+Upgrade

This PR is to add the workflow for DD4hep with DB geometry, however, the workflow is disable for now.

**Comment by chayanit** [comment]
+1

**Comment by silviodonato** [comment]
+1

> Reco comparison results: 5159 differences found in the comparisons

this is not related to this PR, see #34409.

**Pull request merged**